clang-tools-extra/clang-tidy/portability/AvoidPlatformSpecificFundamentalTypesCheck.cpp

```diff
@@ -0,0 +1,260 @@
+    const BuiltinType *BT, ASTContext &Context) const {
+  const TargetInfo &Target = Context.getTargetInfo();
+
+  auto GetReplacementType = [](unsigned Width) {
+    switch (Width) {
+    // This is ambiguous by default since it could be bfloat16 or float16
+    case 16U:
+      return "";
+    case 32U:
+      return "float32_t";
+    case 64U:
+      return "float64_t";
+    case 128U:
+      return "float128_t";
+    default:
+      return "";
+    }
+  };
+
+  switch (BT->getKind()) {
+  // Not an ambiguous type
+  case BuiltinType::BFloat16:
+    return "bfloat16_t";
+  case BuiltinType::Half:
+    return GetReplacementType(Target.getHalfWidth());
+  case BuiltinType::Float:
+    return GetReplacementType(Target.getFloatWidth());
+  case BuiltinType::Double:
+    return GetReplacementType(Target.getDoubleWidth());
+  default:
+    return "";
+  }
+}
+
+void AvoidPlatformSpecificFundamentalTypesCheck::registerMatchers(
+    MatchFinder *Finder) {
+  // Build the list of type strings to match
+  std::vector<std::string> TypeStrings;
+
+  // Add integer types if the option is enabled
+  if (WarnOnInts) {
+    TypeStrings.insert(TypeStrings.end(), {"short",
+                                           "short int",
+                                           "signed short",
+                                           "signed short int",
+                                           "unsigned short",
+                                           "unsigned short int",
+                                           "int",
+                                           "signed",
+                                           "signed int",
+                                           "unsigned",
+                                           "unsigned int",
+                                           "long",
+                                           "long int",
+                                           "signed long",
+                                           "signed long int",
+                                           "unsigned long",
+                                           "unsigned long int",
+                                           "long long",
+                                           "long long int",
+                                           "signed long long",
+                                           "signed long long int",
+                                           "unsigned long long",
+                                           "unsigned long long int"});
+  }
+
+  // Add float types if the option is enabled
+  if (WarnOnFloats) {
+    TypeStrings.insert(TypeStrings.end(),
+                       {"half", "__bf16", "float", "double", "long double"});
+  }
+
+  // Add char types if the option is enabled
+  if (WarnOnChars) {
+    TypeStrings.insert(TypeStrings.end(),
+                       {"char", "signed char", "unsigned char"});
+  }
+
+  // If no types are enabled, return early
+  if (TypeStrings.empty()) {
+    return;
+  }
```

Comment:
```suggestion
  if (TypeStrings.empty())
    return;
```
https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements. There are other places too, please correct.

You could enable https://clang.llvm.org/docs/ClangFormatStyleOptions.html#removebracesllvm in your clang-format file.